src/app/public/modules/phone-field/phone-field-input.directive.ts

```diff
@@ -0,0 +1,247 @@
+   * and the directive name doesn't describe the property.
+   */
+  // tslint:disable-next-line:no-input-rename
+  @Input('skyPhoneFieldInput')
+  public skyPhoneFieldComponent: SkyPhoneFieldComponent;
+
+  @Input()
+  public set defaultCountry(value: string) {
+    this.skyPhoneFieldComponent.setDefaultCountry(value);
+    this._defaultCountryCode = value;
+  }
+
+  public get defaultCountry() {
+    return this._defaultCountryCode;
+  }
+
+  @Input()
+  public set disabled(value: boolean) {
+    this.skyPhoneFieldComponent.disabled = value;
+    this.adapterService.setElementDisabledState(this.elRef.nativeElement, value);
+    this._disabled = value;
+  }
+
+  public get disabled(): boolean {
+    return this._disabled || false;
+  }
+
+  @Input()
+  public formatModel: boolean = true;
+
+  @Input()
+  public skyPhoneFieldNoValidate: boolean = false;
+
+  @Output()
+  public selectedCountryChange = new EventEmitter<SkyCountryData>();
+
+  private set modelValue(value: string) {
+    this._modelValue = value;
+    this.adapterService.setElementValue(this.elRef.nativeElement, value);
+    if (value) {
+      let formattedValue = value;
+      if (this.formatModel) {
+        formattedValue = this.skyPhoneFieldComponent.formatNumber(value.toString());
+      }
+      this._onChange('+' + this.skyPhoneFieldComponent.selectedCountry.dialCode + ' ' + formattedValue);
+    } else {
+      this._onChange(value);
+    }
+    this._validatorChange();
+  }
+
+  private get modelValue(): string {
+    return this._modelValue;
+  }
+
+  private ngUnsubscribe = new Subject();
+
+  private _defaultCountryCode: string;
+
+  private _disabled: boolean;
+
+  private _modelValue: string;
+
+  public constructor(
+    private adapterService: SkyPhoneFieldAdapterService,
+    private changeDetector: ChangeDetectorRef,
+    private elRef: ElementRef,
+    private injector: Injector
+  ) { }
+
+  public ngOnInit(): void {
+    this.adapterService.addElementClass(this.elRef.nativeElement, 'sky-form-control');
+    if (this.defaultCountry) {
+      this.skyPhoneFieldComponent.selectCountry(this.defaultCountry);
+    }
+    this.adapterService.setElementPlaceholder(this.elRef.nativeElement,
+      this.skyPhoneFieldComponent.selectedCountry.placeholder);
+
+    this.adapterService.setAriaLabel(this.elRef.nativeElement);
+  }
+
+  public ngOnDestroy(): void {
+    this.ngUnsubscribe.next();
+    this.ngUnsubscribe.complete();
+    this.selectedCountryChange.complete();
+  }
+
+  public ngAfterViewInit(): void {
+    this.skyPhoneFieldComponent.selectedCountryChange
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe((country: SkyCountryData) => {
+        // Write the value again to cause validation to refire
+        this.writeValue(this.modelValue);
+        this.adapterService.setElementPlaceholder(this.elRef.nativeElement, country.placeholder);
+        this.selectedCountryChange.emit(country);
```

Comment:
This will still emit even if the country doesn't change. For example, if you click "USA", then click "USA" again, it will still emit. Is that deliberate?

Reply:
Good catch. The input directive does not actually store the selected country so I can't do a comparison there (without adding it being stored for just this purpose). However, I added some logic to the component to watch for this scenario and so the input directive won't ever hit its code in this case. Think that is sufficient?

Reply:
If the end result is the consumer doesn't get the same emitted value twice, then I think that'll do it!